lib/services/local/userpreferences.go

```diff
@@ -83,25 +83,7 @@ func (u *UserPreferencesService) UpsertUserPreferences(ctx context.Context, user
 		return trace.Wrap(err)
 	}
 
-	preferences, err := u.getUserPreferences(ctx, username)
-	if err != nil {
-		if !trace.IsNotFound(err) {
-			return trace.Wrap(err)
-		}
-		preferences = DefaultUserPreferences()
-	}
-
-	// when we update preferences, we send the entire cluster preferences object
-	// or the "global" preferences.
-	if prefs.ClusterPreferences != nil {
-		preferences.ClusterPreferences = prefs.ClusterPreferences
-	} else {
-		preferences.Assist = prefs.Assist
-		preferences.Onboard = prefs.Onboard
-		preferences.Theme = prefs.Theme
-	}
-
-	item, err := createBackendItem(username, preferences)
+	item, err := createBackendItem(username, prefs)
```

Comment:
This was removed because originally, we'd only be sending small chunks of the user preferences to update. Now we send the entire thing (before this PR) so we can just completely overwrite the prefs. Using a range to go through what was sent in the protomessage would skip empty values and therefor not allow us to "empty" our pinned resources. Now that we send the entire thing, it works as expected.